### ISSUE TITLE
Memory and socket  leak, telnet clear be blocked 

### ISSUE THREAD

**Issue by Chubru** [pjsip/pjproject#2916]
### Describe the bug

If telnet connection in cli is closed by client, session object is not to be deleted.

https://github.com/pjsip/pjproject/blob/f395420e3f6674901d3fddac586e814dca083ef4/pjlib-util/src/pjlib-util/cli_telnet.c#L1495-L1497

Suggestion:
Add code between line 1496 and 1497
if (sess )
pj_cli_sess_end_session(&(sess->base));

### Steps to reproduce

Open telnet client 8 times and close without "exit" command. On 9-th time telnet will not be opened.

### PJSIP version

2.11.1

### Context

any OS. Tested under Windws, Linux

### Log, call stack, etc

```shell
WAGO> 20:51:11.958   cli_telnet.c !Failure creating active socket
20:51:11.965   cli_telnet.c  Failure creating active socket
```

**Comment by ycaibb** [comment]
@sauwming Should the below patch apply to other fail cases in method `telnet_sess_on_data_read`?
```

if (sess )
            pj_cli_sess_end_session(&(sess->base));
```

https://github.com/pjsip/pjproject/blob/f395420e3f6674901d3fddac586e814dca083ef4/pjlib-util/src/pjlib-util/cli_telnet.c#L1492-L1498

https://github.com/pjsip/pjproject/blob/f395420e3f6674901d3fddac586e814dca083ef4/pjlib-util/src/pjlib-util/cli_telnet.c#L1507-L1509

**Comment by sauwming** [comment]
After rechecking it, it looks like you're right. I have modified the fix in the new PR #2939.
For the second failure case however, I believe the current code is already correct. Explanation about this in the above PR.
